api/filter/fault.proto

```diff
@@ -0,0 +1,125 @@
+syntax = "proto3";
+
+package envoy.api.v2.filter;
+
+import "api/rds.proto";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/wrappers.proto";
+
+// Delay specification is used to inject latency into the rpc/TCP proxy operations.
+message FaultDelay {
+  enum FaultDelayType {
+    // Fixed delay (step function).
+    FIXED = 0;
+    // Exponential delay.
+    EXPONENTIAL = 1;
+  }
+
+  // Delay type to use (fixed|exponential|..). Currently, only fixed delay (step function) is supported.
+  FaultDelayType type = 1;
+
+  // An integer between 0-100 indicating the percentage of operations/connection requests
+  // on which the delay will be injected.
+  google.protobuf.UInt32Value percent = 2;
+
+  oneof fault_delay_type {
+    // Add a fixed delay before forwarding the operation upstream. See
+    // https://developers.google.com/protocol-buffers/docs/proto3#json for
+    // the JSON/YAML Duration mapping. For HTTP/Mongo/Redis, the specified
+    // delay will be injected before a new request/operation. For TCP
+    // connections, the proxying of the connection upstream will be delayed
+    // for the specified period.
+    google.protobuf.Duration fixed_delay = 3;
+  }
+}
+
+// Abort specification is used to prematurely abort a rpc operation/TCP connection
+// with a pre-specified error code.
+message FaultAbort {
+  // An integer between 0-100 indicating the percentage of requests/operations/connections
+  // that will be aborted with the error code provided.
+  google.protobuf.UInt32Value percent = 1;
+
+  // Applicable only for HTTP connections.
+  oneof error_type {
+    // HTTP status code to use to abort the HTTP request.
+    uint32 http_status = 2;
```

Comment:
`google.protobuf.UInt32Value`?

Reply:
I've been in two minds on this, and have stopped objecting to unwrapped types (and in some cases doing it myself) when either:  it's (1) a 100% compulsory field and we don't need defaulting behavior or (2) it's embedded under something like `oneof` where we can test presence/absence.

It's pretty ugly in proto IMHO that we don't just have aliases for wrapped types that express this, e.g. `optional_uint32`, rather than the full `google.protobuf.UInt32Value`.